### ISSUE TITLE
[Bug Fix] Show sub folder option not recursing through nested folders

### ISSUE THREAD

**Issue by nikolaizombie1** [anufrievroman/waypaper#110]
From what I understand, the show sub folders option is supposed to show images that are deeper than the root directory. However, it does not function as I would understand it to.

**Comment by anufrievroman** [comment]
Really? For me, it seems to work. It is supposed to show pictures in subfolders, but not deeper than 1st or 2nd level (don't remember), to prevent extreme situations when it tries to index the whole file system. For you, it always shows only the root directory?

**Comment by nikolaizombie1** [comment]
> Really? For me, it seems to work. It is supposed to show pictures in subfolders, but not deeper than 1st or 2nd level (don't remember), to prevent extreme situations when it tries to index the whole file system. For you, it always shows only the root directory?

I thought it was broken because the folder structure I have for my wallpapers has a depth of 3. I did some testing on my own branch using the glob module and it performed very well. I'll try and benchmark the difference it can produce on a large set of files. 

How about a fully recursive option? That would allow the depth to be infinite. It would be off by default which allows the user to understand that there is a performance hit on a large set of files

**Comment by anufrievroman** [comment]
Yes, that sounds reasonable, could be one of the ticks in the menu for `Infinite subfolders`, or an option in the config file, like `subfolder_depth = 10` for example.

**Comment by nikolaizombie1** [comment]
> Yes, that sounds reasonable, could be one of the ticks in the menu for `Infinite subfolders`, or an option in the config file, like `subfolder_depth = 10` for example.

How about if the depth is a negative integer, it means that its infinite?

**Comment by anufrievroman** [comment]
Yep, and 0 would be only root folder.

**Comment by anufrievroman** [comment]
Thank you again for the PR. I also added some refinements and and called it "Show all subfolders" just to make it more obvious for the used what is being recursed.
